component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/voiceotp/OneTimePassword.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.authenticator.voiceotp;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException;
+
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * This class contains util methods to generate an OTP.
+ */
+public class OneTimePassword {
+
+    // These are used to calculate the check-sum digits.
+    // 0 1 2 3 4 5 6 7 8 9
+    private static final int[] doubleDigits = {0, 2, 4, 6, 8, 1, 3, 5, 7, 9};
+    private static final Log log = LogFactory.getLog(OneTimePassword.class);
+
+    /**
+     * Returns the token generated with random numbers for given size.
+     *
+     * @param size
+     * @return generated token
+     */
+    public static String getRandomNumber(int size) {
+
+        StringBuilder generatedToken = new StringBuilder();
+        try {
+            SecureRandom number = SecureRandom.getInstance(VoiceOTPConstants.ALGORITHM_NAME);
+            // Generate 20 integers 0..20
+            for (int i = 0; i < size; i++) {
+                generatedToken.append(number.nextInt(9));
+            }
+        } catch (NoSuchAlgorithmException e) {
+            log.error("Unable to find the Algorithm", e);
+        }
+
+        return generatedToken.toString();
+    }
+
+    /**
+     * Calculates a checksum value based on a given number and the number of digits in that number.
+     *
+     * @param num    the number to calculate the checksum for
+     * @param digits number of significant places in the number
+     * @return the checksum of num
+     */
```

Comment:
```suggestion
     * Calculates a checksum value based on a given number and the number of digits in that number.
     *
     * @param num    The number to calculate the checksum for.
     * @param digits  Number of significant places in the number.
     * @return The checksum of the number.
     */
```

Check for similar formatting issues in the PR